src/IIIFPresentation/API/Helpers/PaintedResourceX.cs

```diff
@@ -16,7 +16,8 @@ public static bool HasAsset(this List<PaintedResource>? paintedResources)
     /// </summary>
     public static int GetRequiredNumberOfCanvases(this List<PaintedResource>? paintedResources)
     {
-        if (paintedResources.IsNullOrEmpty()) return 0;
+        // no need to generate a canvas id for canvases that have an id, as we'll use that
```

Comment:
The code doesn't line up with this comment. This comment sounds like if we had 10 paintedResources and 4 had and id we'd still return 6 but that isn't happening.

Question is - _should_ it happen? The "all or nothing" for canvasId is MVP only and caught in validator - would it be simpler to implement as in the comment? Would mean less places need changed when we allow specifying canvasIds for some but not all?